File: src/main/java/priv/bajdcc/LALR1/syntax/automata/nga/NGAEdgeData.java
package priv.bajdcc.LALR1.syntax.automata.nga;

import priv.bajdcc.LALR1.semantic.token.ISemanticAction;
import priv.bajdcc.LALR1.syntax.exp.RuleExp;
import priv.bajdcc.LALR1.syntax.exp.TokenExp;
import priv.bajdcc.LALR1.syntax.handler.IErrorHandler;

/**
 * 非确定性文法自动机边数据
 *
 * @author bajdcc
 */
public class NGAEdgeData {
	/**
	 * 边类型
	 */
	public NGAEdgeType kAction = NGAEdgeType.EPSILON;

	/**
	 * 终结符数据
	 */
	public TokenExp token = null;

	/**
	 * 非终结符数据
	 */
	public RuleExp rule = null;

	/**
	 * 存储序号（-1为无效）
	 */
	public int iStorage = -1;

	/**
	 * 错误处理器
	 */
	public IErrorHandler handler = null;

	/**
	 * 语义动作
	 */
	public ISemanticAction action = null;
}

File: src/main/java/com/bajdcc/LALR1/syntax/automata/nga/NGAEdgeData.kt
package com.bajdcc.LALR1.syntax.automata.nga

import com.bajdcc.LALR1.grammar.semantic.ISemanticAction
import com.bajdcc.LALR1.syntax.exp.RuleExp
import com.bajdcc.LALR1.syntax.exp.TokenExp
import com.bajdcc.LALR1.syntax.handler.IErrorHandler

/**
 * 非确定性文法自动机边数据
 * [type] 边类型
 * [token] 终结符数据
 * [rule] 非终结符数据
 * [storage] 存储序号（-1为无效）
 * [handler] 错误处理器
 * [action] 语义动作
 * @author bajdcc
 */
class NGAEdgeData(var type: NGAEdgeType = NGAEdgeType.EPSILON,
                  var token: TokenExp? = null,
                  var rule: RuleExp? = null,
                  var storage: Int = -1,
                  var handler: IErrorHandler? = null,
                  var action: ISemanticAction? = null)